app/src/main/java/com/example/android/weatherapp/viewmodel/HomeScreenViewModel.kt

```diff
@@ -0,0 +1,107 @@
+package com.example.android.weatherapp.viewmodel
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import com.example.android.weatherapp.WeatherApi
+import com.example.android.weatherapp.models.CurrentWeatherModel
+import com.example.android.weatherapp.models.ForecastWeatherModel
+import com.example.android.weatherapp.models.ThreeHourlyWeatherModel
+import com.example.android.weatherapp.models.WeeklyWeatherModel
+import com.example.android.weatherapp.utilities.Helper
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import retrofit2.HttpException
+
+class HomeScreenViewModel : BaseViewModel() {
+
+    private val viewModelJob = Job ()
+    private val coroutineScope = CoroutineScope(Dispatchers.Main + viewModelJob)
+
+    val currentWeather = mutableListOf<CurrentWeatherModel>()
+    val threeHourlyWeatherForecast = mutableListOf<ThreeHourlyWeatherModel>()
+    val weeklyWeatherForecast = mutableListOf<WeeklyWeatherModel>()
+
+    private val _eventDataUpdated = MutableLiveData<Boolean>()
+    val eventDataUpdated: LiveData<Boolean>
+        get() = _eventDataUpdated
+
+    init {
+      fetchData()
+    }
+
+    private fun fetchData() {
+        getCurrentWeather()
+        getForecast()
+    }
+
+    private fun getForecast() {
+        coroutineScope.launch {
+            val forecastObj = WeatherApi.retrofitService.getFiveDaysWeather("Lahore",
```

Comment:
Please use repositories.

Reply:
Added repositories in next commit of this branch